rust/operator-binary/src/controller.rs

```diff
@@ -949,16 +983,39 @@ fn build_metastore_rolegroup_statefulset(
         }
     }
 
+    let recommended_object_labels = build_recommended_labels(
+        hive,
+        &resolved_product_image.app_version_label,
+        &rolegroup_ref.role,
+        &rolegroup_ref.role_group,
+    );
+    // Used for PVC templates that cannot be modified once they are deployed
+    let unversioned_recommended_labels = Labels::recommended(build_recommended_labels(
+        hive,
+        // A version value is required, and we do want to use the "recommended" format for the other desired labels
+        "none",
+        &rolegroup_ref.role,
+        &rolegroup_ref.role_group,
```

Comment:
We would like to add a helper in stackable-operator to produce recommended labels for "role" level things like this.
We should probably also do the same for things that shouldn't be versioned too. Perhaps they can just become `Option<T>` so that we don't need to add another function.

Leave as is for now, and once that method is available, we can go and update it (before the 25.7 release hopefully).